### PR TITLE
Feature#41/products crud

### PR DESCRIPTION
**Pull request by jairolan**
# Description
What did you implemented/Why did you implemented this:
 - I have added product views
- I have configured the actions in the product controller.
- I have configured the product routes.

### CODE REVIEW

File: local_sales/app/views/products/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  <div class="field py-2">
+    <%= form.label :description %>
+    <%= form.text_field :description %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
+<% end %>
```
Comment (xFerchoVJ) [inline local_sales/app/views/products/_form.html.erb:37]: Add blank line

File: local_sales/app/views/products/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<h1>Editing Product</h1>
+
+<%= render 'form', product: @product %>
+<%= link_to 'Show', @product %> |
+<%= link_to 'Back', products_path %>
```
Comment (xFerchoVJ) [inline local_sales/app/views/products/edit.html.erb:5]: Add blank line

File: local_sales/app/views/products/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        <td><%= link_to 'Edit', edit_product_path(product) %></td>
+        <td><%= link_to 'Destroy', product, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+
+<br>
+
+<%= link_to 'New Product', new_product_path %>
```
Comment (xFerchoVJ) [inline local_sales/app/views/products/index.html.erb:27]: Add blank line

File: local_sales/app/views/products/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<div class="m-4">
+    <h1>New Product</h1>
+    
+    <%= render 'form', product: @product %>
+    <%= link_to 'Back', products_path %>
+</div>
```
Comment (xFerchoVJ) [inline local_sales/app/views/products/new.html.erb:6]: Add blank line

File: local_sales/app/views/products/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<p id="notice"><%= notice %></p>
+
+<p>
+  <strong>Name:</strong>
+  <%= @product.name %>
+</p>
+
+<%= link_to 'Edit', edit_product_path(@product) %> |
+<%= link_to 'Back', products_path %>
```
Comment (xFerchoVJ) [inline local_sales/app/views/products/show.html.erb:9]: Add blank line

**Review by xFerchoVJ — DISMISSED** [review]
I I leave comments about add blank lines, but everything LGTM!

File: local_sales/app/controllers/products_controller.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+		# Use callbacks to share common setup or constraints between actions.
+	  def set_product
+	    @product = Product.find(params[:id])
+	  end
+
+	  # Only allow a list of trusted parameters through.
+	  def product_params
+      params.require(:product).permit(:name, :price, :quantity, :description)
+    end
+end
```
Comment (oramasangel) [inline local_sales/app/controllers/products_controller.rb:52]: be careful with the indentation

File: local_sales/app/views/products/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<p id="notice"><%= notice %></p>
```
Comment (oramasangel) [inline local_sales/app/views/products/index.html.erb:1]: you can remove this, cause we can add it in the application layout

**Review by YilKanda — COMMENTED** [review]
Solve the recommendations that your colleagues have given you, please.

File: local_sales/app/controllers/products_controller.rb [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+	# GET /products/1/edit
+  def edit
+  end
+
+  # POST /products
+  def create
+    @product = Product.new(product_params)
+	  @product.save
+    redirect_to @product, notice: 'Product was successfully created.'
```
Comment (juliocabrera820) [inline local_sales/app/controllers/products_controller.rb:26]: I think a validation is missing, I mean, if a product is saved you can redirect and show a message as well as a product is not saved you may determine other actions

File: local_sales/app/controllers/products_controller.rb [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+	# PATCH/PUT /products/1
+  def update
+    @product.update(product_params)
+  	redirect_to @product, notice: 'Product was successfully updated.'
```
Comment (juliocabrera820) [inline local_sales/app/controllers/products_controller.rb:32]: same here

File: local_sales/app/controllers/products_controller.rb [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  # GET /products/1
+  def show
+  end
+
+  # GET /products/new
+  def new
+	  @product = Product.new
+	end
```
Comment (juliocabrera820) [inline local_sales/app/controllers/products_controller.rb:16]: keep your indentation

File: local_sales/app/controllers/products_controller.rb [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  def update
+    @product.update(product_params)
+  	redirect_to @product, notice: 'Product was successfully updated.'
+  end
+
+	# DELETE /products/1
+  def destroy
+	  @product.destroy
+	  redirect_to products_path, notice: 'Offer was successfully destroyed.'
+  end
```
Comment (juliocabrera820) [inline local_sales/app/controllers/products_controller.rb:39]: keep your indentation

File: local_sales/app/controllers/products_controller.rb [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+		# Use callbacks to share common setup or constraints between actions.
+	  def set_product
+	    @product = Product.find(params[:id])
+	  end
+
+	  # Only allow a list of trusted parameters through.
+	  def product_params
+      params.require(:product).permit(:name, :price, :quantity, :description)
+    end
```
Comment (juliocabrera820) [inline local_sales/app/controllers/products_controller.rb:51]: same here with the indentation

File: local_sales/app/views/products/_form.html.erb [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+        <% product.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field py-2">
+    <%= form.label :nombre %>
+    <%= form.text_field :name %>
```
Comment (juliocabrera820) [inline local_sales/app/views/products/_form.html.erb:16]: keep the same field names

**Review by juliocabrera820 — COMMENTED** [review]
Just keep your indentation

**Review by juliocabrera820 — APPROVED** [review]

**Review by JoseRicardoJimenez — APPROVED** [review]
OK!

**Review by xFerchoVJ — COMMENTED** [review]
LGTM

**Review by xFerchoVJ — APPROVED** [review]

**Pull request merged**